taskmanager/src/Application/Task/Task.php

```diff
@@ -57,6 +57,25 @@ public function getStatus(): Status
         return $this->status;
     }
 
+    public function setStatus(Status $status, User $user): void
```

Comment:
IMO this method is not the best option, as we need to provide additional parameters when we need to only change status. 
Please don't merge this PR, I will try to create another PR with different solution.